### DISCUSSION TITLE
v1.6.0

### DISCUSSION THREAD

**Discussion by piksel** [containrrr/watchtower#1778] [Releases]
### Features
8e3bde7 feat: add --health-check command line switch (#1725)
79ebad0 feat: allow logging output to use JSON formatter (#1705)
650acde feat: add a label take precedence argument (#1754)
bbbe041 feat: add no-pull label for containers (#1574)
2e643ed feat(notifications): update shoutrrr to [v0.8](https://github.com/containrrr/shoutrrr/releases/tag/v0.8.0) (#1737)
9f60766 feat: enabled loading http-api-token from file (#1728)
0a5bd54 feat(clean): log removed/untagged images (#1466)
dca45f5 feat: support container network mode (#1429)

### Fixes
bba9b2b fix: empty out the aliases on recreation (#1699)
30280e3 fix(notifications): correctly set the delay from options (#1726)
897b171 fix: only remove container id network aliases (#1724)
dd1ec09 fix: always use container interface (#1516)
25fdb40 fix(registry): image name parsing behavior (#1526)
e8affe3 fix: received typo (#1765)
cfcbcac fix: remove logging of credentials (#1534)
4d661bf fix(registry): ignore empty challenge fields (#1626)
9470bf8 fix: always add missing slashes to link names (#1588)

### Misc
a0fe4a4 ci: add dependabot auto approve
9b28fbc chore: update Dockerfile.dev-self-contained to allow better build cache (#1755)
139f672 test(flags): ensure temp files are cleaned up (#1727)
a56b9bd add testwill as a contributor for doc (#1766)

### Docs
6ace7bd Update notifications.md (#1565)
9180e95 feat(docs): add template preview (#1777)
d1f58c5 docs: clarify what volumes are removed when requested (#1738)
1d5a8d9 test: check flag/docs consistency (#1770)
11423d1 docs: update interval text in introduction (#1767)
b801b63 docs: bump go version in credential helper example (#1749)
32204a7 docs: fix env variable name `notification(s)-delay` (#1486)
b27bd05 docs: add "HTTP API Mode" link to nav menu (#1645)
aa50d12 fix(docs): fix anchor links in the metrics part (#1522)

<details>
<summary>

### Dependencies

</summary>

bde1383 Merge pull request #1548 from containrrr/dependabot/go_modules/github.com/onsi/gomega-1.26.0
7de7356 Merge pull request #1549 from containrrr/dependabot/github_actions/goreleaser/goreleaser-action-4.2.0
7648e9c chore(deps): bump github.com/docker/docker from 24.0.5+incompatible to 24.0.6+incompatible (#1761)
6685fef chore(deps): bump github.com/docker/cli from 24.0.5+incompatible to 24.0.6+incompatible (#1762)
856fd25 chore(deps): bump golang.org/x/net from 0.14.0 to 0.15.0 (#1760)
da39883 chore(deps): bump goreleaser/goreleaser-action from 4.4.0 to 5.0.0 (#1759)
36391b0 chore(deps): bump actions/checkout from 3 to 4 (#1750)
cd458fa chore(deps): bump golang.org/x/text from 0.12.0 to 0.13.0 (#1751)
0d2eba1 chore(deps): bump goreleaser/goreleaser-action from 4.3.0 to 4.4.0 (#1733)
02da45d chore(deps): bump alpine from 3.18.2 to 3.18.3 in /dockerfiles (#1732)
7948242 chore(deps): bump golang.org/x/net from 0.12.0 to 0.14.0 (#1720)
be113d7 chore(deps): bump golang.org/x/text from 0.11.0 to 0.12.0 (#1721)
a5d7f23 chore(deps): bump github.com/docker/cli (#1701)
5eb00cc chore(deps): bump github.com/docker/docker (#1702)
7dc8d9f chore(deps): bump github.com/onsi/gomega from 1.27.8 to 1.27.10 (#1700)
dfe4346 chore(deps): bump github.com/docker/cli from 24.0.2+incompatible to 24.0.4+incompatible (#1695)
32988aa chore(deps): bump golang.org/x/net from 0.11.0 to 0.12.0 (#1692)
1b3a5d7 chore(deps): bump github.com/docker/docker from 24.0.2+incompatible to 24.0.4+incompatible (#1694)
787ce72 chore(deps): bump golang.org/x/net from 0.10.0 to 0.11.0 (#1678)
244e3ce chore(deps): bump github.com/prometheus/client_golang from 1.15.1 to 1.16.0 (#1677)
243b217 chore(deps): bump alpine from 3.18.0 to 3.18.2 in /dockerfiles (#1679)
c7c1aee chore(deps): bump github.com/spf13/viper from 1.15.0 to 1.16.0 (#1667)
170c79d chore(deps): bump github.com/sirupsen/logrus from 1.9.2 to 1.9.3 (#1666)
a7ca783 chore(deps): bump docker/login-action from 2.1.0 to 2.2.0 (#1672)
ad644fc chore(deps): bump github.com/onsi/gomega from 1.27.7 to 1.27.8 (#1671)
b0acc8f chore(deps): bump goreleaser/goreleaser-action from 4.2.0 to 4.3.0 (#1673)
4495445 chore(deps): bump golang.org/x/text from 0.9.0 to 0.10.0 (#1670)
1180691 chore(deps): bump github.com/stretchr/testify from 1.8.3 to 1.8.4 (#1668)
6c9dd50 chore(deps): bump github.com/docker/docker from 23.0.6+incompatible to 24.0.2+incompatible (#1663)
1fed2a8 chore(deps): bump github.com/docker/cli from 24.0.1+incompatible to 24.0.2+incompatible (#1664)
4a5c038 chore(deps): bump github.com/docker/cli (#1661)
89da17a chore(deps): bump github.com/stretchr/testify from 1.8.2 to 1.8.3 (#1660)
9806c95 chore(deps): bump github.com/onsi/gomega from 1.27.6 to 1.27.7 (#1658)
bac02e7 chore(deps): bump github.com/sirupsen/logrus from 1.9.0 to 1.9.2 (#1659)
0a74e50 chore(deps): bump alpine from 3.17.3 to 3.18.0 in /dockerfiles (#1653)
021e9f3 chore(deps): bump github.com/docker/distribution from 2.8.1+incompatible to 2.8.2+incompatible (#1652)
43a296a chore(deps): bump github.com/prometheus/client_golang (#1647)
e271286 chore(deps): bump github.com/docker/cli (#1649)
0802926 chore(deps): bump golang.org/x/net from 0.9.0 to 0.10.0 (#1648)
c7499e8 chore(deps): bump github.com/docker/docker (#1650)
47dcb49 chore(deps): bump github.com/docker/cli (#1640)
9957fff chore(deps): bump github.com/docker/docker from 23.0.4+incompatible to 23.0.5+incompatible (#1641)
44436eb chore(deps): bump github.com/docker/docker from 23.0.3+incompatible to 23.0.4+incompatible (#1635)
311bb93 chore(deps): bump github.com/docker/cli from 23.0.3+incompatible to 23.0.4+incompatible (#1634)
aec7762 chore(deps): bump github.com/prometheus/client_golang (#1630)
9d6b008 chore(deps): bump github.com/stretchr/testify from 1.8.1 to 1.8.2 (#1621)
a34c02f chore(deps): bump github.com/robfig/cron (#1590)
46f24d2 chore(deps): bump github.com/spf13/cobra from 1.6.1 to 1.7.0 (#1622)
288ed11 chore(deps): bump github.com/onsi/gomega from 1.26.0 to 1.27.6 (#1623)
3f091f1 chore(deps): bump golang.org/x/net from 0.5.0 to 0.9.0 (#1620)
d0617aa chore(deps): bump golang.org/x/text from 0.6.0 to 0.8.0 (#1575)
dfcaf07 chore(deps): bump github.com/docker/cli from 20.10.23+incompatible to 23.0.3+incompatible (#1618)
035572f chore(deps): bump github.com/docker/docker from 23.0.2+incompatible to 23.0.3+incompatible (#1617)
81d2376 chore(deps): bump actions/setup-go from 3 to 4 (#1596)
c6c01c8 chore(deps): bump alpine from 3.17.1 to 3.17.3 in /dockerfiles (#1613)
df1b86b chore(deps): bump docker/docker from 20.10.23+inc to 23.0.2+inc (#1612)
ee8f293 chore(deps): bump andrewslotin/go-proxy-pull-action from 1.0.3 to 1.1.0 (#1555)
b94741d chore(deps): bump goreleaser/goreleaser-action from 4.1.0 to 4.2.0
98d0c47 chore(deps): bump github.com/onsi/gomega from 1.25.0 to 1.26.0
</details>

## Docker images

- `docker pull containrrr/watchtower:i386-1.6.0`
- `docker pull containrrr/watchtower:i386-latest`
- `docker pull ghcr.io/containrrr/watchtower:i386-1.6.0`
- `docker pull ghcr.io/containrrr/watchtower:i386-latest`
- `docker pull containrrr/watchtower:arm64v8-1.6.0`
- `docker pull containrrr/watchtower:arm64v8-latest`
- `docker pull ghcr.io/containrrr/watchtower:arm64v8-1.6.0`
- `docker pull ghcr.io/containrrr/watchtower:arm64v8-latest`
- `docker pull containrrr/watchtower:armhf-1.6.0`
- `docker pull containrrr/watchtower:armhf-latest`
- `docker pull ghcr.io/containrrr/watchtower:armhf-1.6.0`
- `docker pull ghcr.io/containrrr/watchtower:armhf-latest`
- `docker pull containrrr/watchtower:amd64-1.6.0`
- `docker pull containrrr/watchtower:amd64-latest`
- `docker pull ghcr.io/containrrr/watchtower:amd64-1.6.0`
- `docker pull ghcr.io/containrrr/watchtower:amd64-latest`

<hr /><em>This discussion was created from the release <a href='https://github.com/containrrr/watchtower/releases/tag/v1.6.0'>v1.6.0</a>.</em>

**Comment by jdb9294** [comment]
Not sure what was going on, but Watchtower was creating a new container every poll cycle.  Docker was unable to start the container because the ports were already in use by the existing container.  I have my poll set to 3600, so I had about 18 containers created this morning.

I think I resolved it by just removing all the watchtower containers and redeploying from the compose file.  Just not sure if this was a bug in 1.6.0 or an issue with my configuration.

Worth noting that I have 3 separate docker servers, all three had the same issue. 

My docker compose:
```
version: "3"
services:
  watchtower:
    container_name: watchtower_management
    hostname: watchtower_management
    image: containrrr/watchtower
    restart: unless-stopped
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    environment:
      TZ: America/New_York
      WATCHTOWER_POLL_INTERVAL: 3600
      WATCHTOWER_NO_STARTUP_MESSAGE: false
      WATCHTOWER_CLEANUP: true
      WATCHTOWER_MONITOR_ONLY: false
      WATCHTOWER_NOTIFICATIONS: email
      WATCHTOWER_NOTIFICATION_EMAIL_FROM: ***@***.***
      WATCHTOWER_NOTIFICATION_EMAIL_TO: ***@***.***
      WATCHTOWER_NOTIFICATION_EMAIL_SERVER: ***.***
      WATCHTOWER_NOTIFICATION_EMAIL_SERVER_PORT: 25
      WATCHTOWER_NOTIFICATION_EMAIL_DELAY: 2
      WATCHTOWER_HTTP_API_METRICS: true
      WATCHTOWER_HTTP_API_TOKEN: ***
    ports:
      - mode: ingress
        target: 8080
        published: "8585"
        protocol: tcp
```

**Comment by C0ll1nDr4k3** [comment]
That JSON formatter is going to be a lifesaver.

**Reply by C0ll1nDr4k3** [reply]
Message Pack support next?
